src/modules/help.ts

```diff
@@ -0,0 +1,85 @@
+import { command, default as CookiecordClient, Module } from 'cookiecord';
+import { Message, MessageEmbed } from 'discord.js';
+
+export default class HelpModule extends Module {
+	protected client: any;
+	constructor (client: CookiecordClient) {
+		super(client);
+
+		this.client = client;
+	}
+
+	@command({ aliases: ['help', 'commands', 'h'] })
+	async help (message: Message, args: string) {
+		if (!args) {
+			let embed = new MessageEmbed()
+				.setAuthor(
+					message.guild.name,
+					message.guild.iconURL({ dynamic: true }),
+				)
+				.setTitle('Help Command!')
+				.setDescription(
+					`Hello ${message.author.username}! Here is a list of all commands in me! To get detailed description on any specific command, do \`help <command>\``,
+				)
+				.addField(
+					`**Misc Commands:**`,
+					`\`help\` ► View a list of all commands!\n\`ping\` ► View the latency of the bot\n\`ask\` ► Sends a message with a link redirecting to dontasktoask.com\n\`playground\` ► Just enter your code and get a link with TS playground!`,
+				)
+				.addField(
+					`**Help Channel Commands:**`,
+					`\`done\` ► Close a __ongoing__ help channel opened by you!`,
+				)
+				.addField(
+					`**Reputation Commands**`,
+					`\`rep\` ► Did somebody help you? Give them a reputation point!\n\`history\` ► Check the reputation history of a user!\n\`leaderboard\` ► See the reputation leaderboard!`,
+				)
+				.setFooter(
+					this.client.user.username,
+					this.client.displayAvatarURL(),
+				)
+				.setTimestamp();
+
+			return message.channel.send(embed);
+		} else {
+			switch (args[0]) {
```

Comment:
You can automatically generate this by using the description field in all the `@command`'s and then getting the command from Cookiecord based on the user input, look at `CookiecordClient#commandManager#getByTrigger`